packages/astro/src/runtime/server/render/component.ts

```diff
@@ -375,6 +375,15 @@ If you're still stuck, please open an issue on GitHub or join us at https://astr
 				}
 			}
 			destination.write(createRenderInstruction({ type: 'directive', hydration }));
+			if (hydration.directive !== 'only' && renderer?.ssr.renderHydrationScript) {
+				destination.write(
+					createRenderInstruction({
+						type: 'renderer-hydration',
+						rendererName: renderer.name,
+						render: renderer?.ssr.renderHydrationScript,
```

Comment:
I don't think the optional chaining operator `?` is needed, because `render` is a mandatory property